RST-1346-et1-welsh-translation

sivvoy opened the pull request.

Commit:
et1 english and welsh translation

Commit:
et1 apply page

Commit:
test to cover You feedback link

Commit:
add instructions on how to run different languages

Commit:
valiate apply page in welsh langugage

Commit:
fix broken test

Commit:
remove as its no longer needed

Commit:
Update README.md

Commit:
rename feature test

Commit:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

garytaylor commented on features/et1/et1_apply_page_in_welsh_translation.feature:
```diff
@@ -0,0 +1,26 @@
+  Background: Apply Page in Welsh
+    Given a claimant is on the Apply page in 'Welsh' 
+
+  Scenario: Your feedback
```
I think this will be slow as each scenario will be fetching the page again - and whilst the apply page is the first page, when you get round to testing something which is say 6 pages in, you wont want to be doing separate scenarios for different bits of content like this ?

garytaylor commented on features/support/capybara_selectors/paragraph_link_named.rb:
```diff
@@ -0,0 +1,6 @@
+Capybara.add_selector(:paragraph_link_named) do
```
This selector seems to be nothing to do with a link ?  it is more like 'paragraph_containing' or something like that ?

garytaylor commented on test_common/page_objects/et1/new_claim_page.rb:
```diff
@@ -3,15 +3,62 @@ module EtFullSystem
         end
-
+        element :header, :h1_text, 'claims.new.header', exact: false
+        element :are_you_in_time_h2, :h2_text, 'claims.new.are_you_in_time.legend', exact: false
```
Minor one - but the element names should be meaningful - no need to put things like 'h2' in them - as that is more the 'technical boring stuff' that your test suite probably doesnt care about - it just wants the 'are you in time' paragraph or something along those lines ?

garytaylor reviewed (approved):
All looks good - made a few comments but nothing that should stop this being merged.  Just be a bit careful about the performance side with the way you are loading the page multiple times etc..

Commit:
add apply page in english translation

Commit:
based on gary's comment rename selector and validate all copy text in one scenario'

Commit:
rename selectors to make it more meaningful

Commit:
remove duplicated test

Commit:
Update README.md

Commit:
remove duplidated test

Commit:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
add test to cover links to other pages

Commit:
Merge branch 'develop' into rst-1346-et1-welsh-translation

Commit:
fix broken test

Commit:
add saving your claim translation test

Commit:
add test to cover print this page

Commit:
add todo test scenario for when i have time

Commit:
add test to cover return your claim page

Commit:
add test to cover saved claimant

Commit:
rename page object to match the page name in the url as its very confusing if not

Commit:
add test to cover start a new claim

Commit:
remove the word translation as the test should work in both english and en

Commit:
add your feedback page

Commit:
add test to submit your feedback

Commit:
clear browser session so that it can reload with a new session

Commit:
add test to cover time limits

Commit:
add test to validate guide certificate number

Commit:
add missing text

Commit:
rename page object based on the page url

Commit:
read from translator

Commit:
allow claimant details page to handle switching languages

Commit:
point to the right branch

Commit:
Merge branch 'develop' into rst-1346-et1-welsh-translation

Commit:
add checks to validate claimant details in english

Commit:
add checks to validate claimant details in welsh

Commit:
add checks to valudate assistance need translation

Commit:
add test to validate do i need assistant

Commit:
removed unused code

Commit:
check for country values

Commit:
validate error message for claimant's details page in welsh and in english

Commit:
validation against blank email address

Commit:
fix broken test

Commit:
validate invalid email address

Commit:
fix broken test

Commit:
add test for invalid email address

Commit:
fix broken test

Commit:
change test to wait for element or sections to be represent

Commit:
change test to wait for element or sections to be represent

Commit:
fixed welsh version for contact preference

garytaylor committed:
Added factory translation to the claimant title as an example

garytaylor committed:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

# Conflicts:
#	test_common/page_objects/et1/claimant_details_page.rb

Commit:
setup factorybot translator

Commit:
add test to fill in all fields on the claimant details page

Commit:
fix country as it was working properly

Commit:
add group claims test

Commit:
add invalid uk postcode error message

Commit:
add page object for additional claimants and upload

Commit:
verify copy text for group claimants

Commit:
rename it back group claimants

Commit:
remove unused code

Commit:
submit without no claimants

Commit:
add page object for representatives

Commit:
verify copy text for group claimants in welsh and in english

Commit:
trying to multiple claimants but unable to

garytaylor committed:
Fix to group claimants page

Commit:
refactor test to get it working in english and in welsh for group claimants

Commit:
verify copy text for csv upload file

Commit:
fixed scoping issues around dob

Commit:
fix upload csv file

Commit:
add test to validate invalid dob

Commit:
add test for representative

Commit:
fix broken test for claimant

Commit:
fix broken test so that gary and stath can branch of it\

Commit:
verify copy text for representative details page

Commit:
when answering no to representative then user is taken to the respondent details page

Commit:
submit blank representative questions

Commit:
validate invalid postcode on representative page

Commit:
representative details page can now run in english and welsh version

Commit:
remove e2e tagging from et1 folder

Stivaros committed:
Add page object and basic feature for landing page

Stivaros committed:
Add page object and basic feature for Claim Type

Commit:
test to cover respondent details page

garytaylor committed:
Added PDF validation

garytaylor committed:
Added PDF validation

Commit:
test additional respondents details page

Commit:
fix merging issues

Commit:
fix broken test by reverting to old code

Commit:
revert back to old code so that additional claimant test wont break

Commit:
Merge pull request #99 from ministryofjustice/rst-1346-gary

Rst 1346 gary

Stivaros committed:
Add page object and basic feature for Sexual Identity

Commit:
additional respondents can now handle english and welsh translation

Commit:
add test for employment details page

Commit:
employment details translation

Commit:
proceed to about the claim page when answering no to employment details

Commit:
complete flow for employment details

Commit:
complete flow for employment details

Commit:
claim type page for translation

Stivaros committed:
Add page object and basic feature for Relationship Status

Stivaros committed:
Add page object and basic feature for Age Group Caring Responsibilities

Stivaros committed:
Add page object and basic feature for Religion

Commit:
add test to select all claim types and validate error message if claim type is not selected

Stivaros committed:
Add page object and basic feature for Ethnic Group

Stivaros committed:
Add page object and basic feature for Health Page

Stivaros committed:
Add page object and basic feature for Pregnancy Page

Commit:
add translation for claim details page

Commit:
fix problem test

Stivaros committed:
Update ET1 with Diversity translations

Commit:
add translation for claim outcome page

Commit:
fix broken test

Stivaros committed:
Fix Pregnancy feature and step names

Stivaros committed:
Add page object for Submission Form Page

Stivaros committed:
Add page object for Confirmation Page

Stivaros committed:
Fix set-url on Diversity page objects

Stivaros committed:
Remove empty rows in page objects

Stivaros committed:
Use translated strings to identify buttons

Commit:
add translation for more about the claim page

Commit:
fix broken test for diversity

garytaylor committed:
Submission page tests added and page objects complete with exception of the employment section as we are not exercising that thoroughly in this test suite yet

garytaylor committed:
Removed localize method from i18n stuff

Commit:
point to the latest et1 system

Commit:
Merge pull request #101 from ministryofjustice/rst1346-gary

Submission page tests added and page objects complete with exception …

Commit:
update submission form with the correct translation

garytaylor committed:
Code for siv to review - accessing the admin API via admin page objects

garytaylor committed:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
add logic to handle switching between english and welsh in et1 and diveristy pages

Commit:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
fix broken test

Commit:
map against latest yml file

Commit:
fix broken test

Commit:
fix other religion free text field

Commit:
fix translation issue for ethnicity

Commit:
fix broken test

Commit:
add add test to validate diversity review page

Commit:
fix diversity for welsh translation

garytaylor committed:
Massive sort out for pdf section file objects - now reports errors much better.
Now highlighting real errors with the  welsh pdf

garytaylor committed:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
validate claim submitted page in welsh and in english

Commit:
test for unknown address

Commit:
validate rtf and csv files

Commit:
fix et1 txt format as factory_bot  work address changed

Commit:
remove duplidate test

Commit:
remove plural in claimants, respondents and representatives

Commit:
fix broken test

Commit:
fix diversity link

Commit:
remove space as it was causing the test to failed

garytaylor committed:
Now requiring base section ahead of sections in pdf

garytaylor committed:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

garytaylor committed:
Updated et1 to head of 1346 branch

garytaylor committed:
Hopefully sorted out the missing method error in some environments for the pdf section

Commit:
fix et1 email notification to get ti working in english

garytaylor committed:
Trying require_relative for base sections for pdf

Commit:
fix broken et3 email notification due to changes in factorybot

Commit:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
add et3 welsh translator

garytaylor committed:
Removed duplicate definition of representative_contact_preference in factories

Commit:
fix email notfication in welsh

Commit:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
fix broken claim submitted in welsh

garytaylor committed:
Improving pdf file sections

garytaylor committed:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

Commit:
replace have_answer with eq to make the test pass

garytaylor committed:
Improved rspec eventually matcher to report true number of attempts

garytaylor committed:
The test with N/A in the name is now working - as the time is added to the last name

garytaylor committed:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

garytaylor committed:
Fix to employment details section in pdf tests

garytaylor committed:
Attempt at parallel running

garytaylor committed:
Added empty gitkeep in tmp folder

garytaylor committed:
Fixes to pdf file objects

garytaylor committed:
Fixes to factories to use et1 information instead of et3 information for respondents
Et1 export helper now only uses english translation as all multi selection options should go to ethos as english

garytaylor committed:
factory_translate method can now pass extra arguments on to the underlying t method

garytaylor committed:
Tidy up

garytaylor committed:
Tidy up

Commit:
fix factorybot which was causing the test to failed

garytaylor committed:
All pdf values are now unescaped - as they end up http entity encoded inside the pdf - and they come out like that from pdftk

garytaylor committed:
Corrected 1 last et3_information - should have been et1_information

garytaylor committed:
Replaced all delegate methods to have a real method - next version of site prism breaks using active support's delegate method as the sections use ruby's Forwardable module

garytaylor committed:
Employment details page corrected - had a scope issue which breaks after version 2.12 of site prism
This is because site prism version < 2.12 has a bug where the scope is not used in sections - 2.12 and after fixes this bug, highlighting scoping issues like this one.
The legend header was finding the wrong element.

garytaylor committed:
Bug fix - missed this one after the employment page change

Commit:
add timeout to flaky test

Commit:
use et1 factorybot

Commit:
use et1 data

Commit:
Merge branch 'rst-1346-et1-welsh-translation' of github.com:ministryofjustice/et-full-system into rst-1346-et1-welsh-translation

garytaylor committed:
Updated site capybara and cucumber as far as they will go
Removed video recording

Commit:
Merge pull request #103 from ministryofjustice/upgrade-test-environment

Updated site capybara and cucumber as far as they will go

Commit:
point to the latest repos

garytaylor committed:
Merge remote-tracking branch 'origin/develop' into rst-1346-et1-welsh-translation
